source/audio/Music.cpp

```diff
@@ -217,16 +212,16 @@ void Music::Decode()
 				memcpy(&input.front(), stream.next_frame, remainder);
 
 			// Now, read a chunk of data from the file.
-			size_t read = fread(&input.front() + remainder, 1, INPUT_CHUNK - remainder, file);
+			file->read(reinterpret_cast<char *>(input.data() + remainder), INPUT_CHUNK - remainder);
 			// If you get the end of the file, loop around to the beginning.
-			if(!read || feof(file))
-				rewind(file);
+			if(file->eof())
+				file->seekg(0, iostream::beg);
```

Comment:
(see one of my previous comments)
```suggestion
				file->seekg(0, ios::beg);
```